benchmarks/src/main/java/org/elasticsearch/benchmark/index/mapper/MapperServiceFactory.java

```diff
@@ -71,7 +71,8 @@ public static MapperService create(String mappings) {
                 public <T> T compile(Script script, ScriptContext<T> scriptContext) {
                     throw new UnsupportedOperationException();
                 }
-            }
+            },
+            /*EmptyMeteringCallback.INSTANCE*/null
```

Comment:
ignore the EmptyMeteringCallback. It should be DocumentReporterFactory
the question here is - should we export the `org.elasticsearch.plugins.internal.documentreporting` to benchmarks? otherwise it won't compile

Reply:
Why do benchmarks need this?

Reply:
they need MapperService so perhaps I should create a new constructor in MapperService which will default to EMPTY_INSTANCE?

Reply:
If you need to export it to benchmarking I think that is ok.

Reply:
I will consult this with Chris next week. Perhaps we could use adds-exports or tweak the gradle build somehow.
 benchmarks does not use module at the moment. It is just an annoying intellij error

Reply:
I have added an add-exports https://github.com/elastic/elasticsearch/pull/97961/files#diff-77ab8968a72299a5fe665cc05d74e3f4c8d3059b243f8fc52d94462a898909fbR51 but this still shows up as red in intellij. 
this might be a bug in intellij https://youtrack.jetbrains.com/issue/IDEA-154038/IDEA-doesnt-respect-Gradle-compiler-settings unless @breskeby knows how to force intellij to use gradle defined compiler settings?

Noone is running benchmarks from intellij anyway.